sources/engine/Stride.VirtualReality/OpenVR/OpenVR.cs

```diff
@@ -16,18 +16,6 @@ namespace Stride.VirtualReality
 {
     internal static class OpenVR
     {
-        /// <summary>Bypasses definite assignment rules for a given reference.
-        /// <para>A thin wrapper around <see cref="Unsafe.SkipInit{T}(out T)"/> for the sole purpose of making it usable in an expression.</para></summary>
-        /// <typeparam name="T">The type of the reference.</typeparam>
-        /// <param name="value">The reference whose initialization should be skipped.</param>
-        /// <returns>The reference to <paramref name="value"/>.</returns>
-        /// <remarks>Take care to ensure that the struct has been initialized appropriately, otherwise the struct's fields could contain uninitialized data from the stack.</remarks>
-        private static unsafe ref T SkipInit<T>(out T value)
-        {
-            Unsafe.SkipInit(out value);
-            return ref value;
-        }
-
```

Comment:
`warning CS9088: This returns a parameter by reference 'value' but it is scoped to the current method`

Reply:
Would be good to know the reason why this was used. @xen2 any idea?

Reply:
Right, I didn't specify, but this is fairly obvious when you look at the callsites.
The issue this method was solving was unsafe as cast with an uninitialized out value, i.e. this, is illegal in c#:
```cs
void MyF(out int value) => Unsafe.As<float, int>(ref value) = 0.1f;
```
The out value has to be assigned before usage as ref parameters.
So, this method is used to make the compiler think it was initialized to avoid the added cost of actually initializing it when the value would be initialized through other means (in this case, through the ref returned by `Unsafe.As`).

Now, the warning this was generating is related to the scopes of ref return values, the ref may leak depending on how it is used by the callsite, pointing at some undefined address location. In this case it couldn't leak since everything happened in the same scope, but it is trivial to solve the issue without using this fairly unsafe method - we can just convert the value you would assign to the out parameter instead of converting the out parameter itself, like I have done below in the same file.

Reply:
Interestingly, the JIT is different (albeit equivalent) for both versions. And surprisingly, the version that used SkipInit did actually do a memset (because it is vectorized for some reason). [See on SharpLab](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBDAzgWwB8ABAJgEYBYAKGIAYACY8gOgBEBLbAcwDsJcGDmFwBuGvSasASgFdeQ/DBYBhCPgAOHADYwoAZT0A3YTDE0JjfQAtsUDQBlswFnIUcl46hIDMTUgwqDADeNAzhDGERglCyYBgMAGrY2lHhodQRWUx+AGbaENgJuV5ZAL5pkZkRGlAcRkUwSSkMRgwAvAy8MADuzdohDLkdDHQsAKzDZaURlcR+xCgMAPLaACYAFBCyCRwKrSmyMACUlRnZEQCqvLjYucoAgrgAPHsYaEMFRQB8G7DD+gA1hwNABJXgcDBbHYHbRHY7HEZGFglSoVarhOYLJarNakaG7fYNOEnM6VLLXW73FhAkHgyEE2HwmYXSl3R4vN4ffKFDC/f5Mk5IlEshjorJYphLAByvUZb0FpwxIXJEWJRxGbOpT2ePKKHze/Jgw2RuWOovFs2VAG0AFKQgDiMG6dTAGwwAE8NDAILkNm8ER97RgnS7hO6vT6/XqMAiALqVWr1RpSJAMeRUpoCgAqDFpYIhGGe2d+2wSufVpOV5wuDC1ynz9KhZcVoqyxAA7AwBZWLTQykA==).